qa/tasks/cephfs/test_misc.py

```diff
@@ -129,6 +130,29 @@ def test_evict_client(self):
         ls_data = self.fs.mds_asok(['session', 'ls'])
         self.assert_session_count(1, ls_data)
 
+    def test_unmount_for_evicted_client(self):
+        """Test if client hangs on unmount after evicting the client."""
+        # Only one client is required for this test.
+        self.mount_b.umount()
+        self.mount_b.umount_wait()
+
+        self.fs.mon_manager.raw_cluster_cmd('fs', 'set', self.fs.name,
+                                            'allow_multimds', 'true',
+                                            '--yes-i-really-mean-it')
+        self.fs.set_max_mds(2)
+        self.fs.wait_for_daemons()
+
+        output = self.fs.mon_manager.raw_cluster_cmd('tell', 'mds.0', 'client',
+                                                     'ls')
+        client_id = output[output.find('"id": ') + 6 :
+                           output.find(',', output.find('"id": '))]
+        self.fs.mon_manager.raw_cluster_cmd('tell', 'mds.0', 'client', 'evict',
+                                            'id={0}'.format(client_id))
+
+        # Test: Instead of hanging, client should unmount.
+        self.mount_a.umount()
```

Comment:
This line reveals an issue: the method `umount()` uses `fusermount` to unmount the filesystem. The client unmounts successfully but ceph-fuse remains alive in the background. This makes the method `umount_wait()` (on the next line) wait until the timeout and, as a result, the test fails. I see the same behaviour when I do it manually.

Contradictorily, using umount command leads to both - CephFS vacating the mount point as well as exiting ceph-fuse process. So, must this mean a bug on the Ceph side?

Reply:
> Contradictorily, using umount command leads to both - CephFS vacating the mount point as well as exiting ceph-fuse process. So, must this mean a bug on the Ceph side?

That's interesting. @ukernel @jtlayton do you have an explanation why `umount` would cause ceph-fuse to exit when `fusermount -u` does not?